matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/internal/DefaultPatientMatch.java

```diff
@@ -0,0 +1,497 @@
+    private Long id;
+
+    @Basic
+    private String referencePatientId;
+
+    @Basic
+    private String referenceServerId;
+
+    @Basic
+    private String matchedPatientId;
+
+    @Basic
+    private String matchedServerId;
+
+    @Basic
+    private Timestamp foundTimestamp;
+
+    @Basic
+    private Boolean notified;
+
+    @Basic
+    private Timestamp notifiedTimestamp;
+
+    @Basic
+    private Boolean rejected;
+
+    @Basic
+    private Double score;
+
+    @Basic
+    private Double genotypeScore;
+
+    @Basic
+    private Double phenotypeScore;
+
+    @Basic
+    /*
+     * an href to remote patient. The fields serverId and href are both null or both not null.
+     */
+    private String href;
+
+    // Attributes related to reference patient
+
+    @Basic
+    @Column(columnDefinition = "CLOB")
+    private String genes;
+
+    @Transient
+    private Set<String> genesSet;
+
+    @Basic
+    @Column(columnDefinition = "CLOB")
+    private String phenotypes;
+
+    @Transient
+    private PhenotypesMap phenotypesMap;
+
+    // Attributes related to matched patient
+
+    @Basic
+    @Column(columnDefinition = "CLOB")
+    private String matchedGenes;
+
+    @Transient
+    private Set<String> matchedGenesSet;
+
+    @Basic
+    @Column(columnDefinition = "CLOB")
+    private String matchedPhenotypes;
+
+    @Transient
+    private PhenotypesMap matchedPhenotypesMap;
+
+    @Transient
+    private PatientInMatch referencePatientInMatch;
+
+    @Transient
+    private PatientInMatch matchedPatientInMatch;
+
+    static {
+        DefaultPatientMatchUtils utils = null;
+        try {
+            ComponentManager ccm = ComponentManagerRegistry.getContextComponentManager();
+            utils = ccm.getInstance(DefaultPatientMatchUtils.class);
+        } catch (ComponentLookupException e) {
+            LOGGER.error("Error loading static components: {}", e.getMessage(), e);
+        }
+        UTILS = utils;
+    }
+
+    /**
+     * Hibernate requires a no-args constructor.
+     */
+    public DefaultPatientMatch()
+    {
+    }
+
+    /**
+     * Build a DefaultPatientMatch from a PatientSimilarityView. Both patients are local.
+     *
+     * @param similarityView the object to read match from
+     */
+    public DefaultPatientMatch(PatientSimilarityView similarityView)
+    {
+        this.initialize(similarityView, null, null);
+    }
+
+    /**
+     * Build a DefaultPatientMatch from a PatientSimilarityView.
+     *
+     * @param similarityView the object to read match from
+     * @param referenceServerId id of server where reference patient is found
+     * @param matchedServerId if of server where matched patient is found
+     */
+    public DefaultPatientMatch(PatientSimilarityView similarityView, String referenceServerId, String matchedServerId)
+    {
+        this.initialize(similarityView, referenceServerId, matchedServerId);
+    }
+
+    /**
+     * TODO remove.
+     *
+     * @param testData testData
```

Comment:
It would be helpful to describe the condition in which this code should be removed, then.